packages/babel-core/src/config/full.js

```diff
@@ -278,17 +279,53 @@ const instantiatePlugin = makeWeakCache(
   },
 );
 
+const validateIfOptionNeedsFilename = (
+  options: ValidatedOptions,
+  descriptor: UnloadedDescriptor,
+): void => {
+  if (options.test || options.include || options.exclude) {
+    const formattedPresetName = descriptor.name
+      ? `"${descriptor.name}"`
+      : "/* your preset */";
+    throw new Error(
+      [
+        `Preset ${formattedPresetName} requires a filename be set.`,
```

Comment:
I read again and feel like
```
Preset ${formattedPresetName} requires a filename to be set when babel is called directly,
`\`\`\``,
`babel.transform(code, { filename: 'file.ts', presets: [${formattedPresetName}] });`,
`\`\`\``
```

And we may can change `file.js` to `file.ts` since this is the popular path.

Reply:
> And we may can change file.js to file.ts since this is the popular path

Good idea